module/surveySubmissions/helper.js

```diff
@@ -0,0 +1,424 @@
+   * @param {Array} [surveySubmissionFilter = "all"] - survey submission ids.
+   * @param {Array} [fieldsArray = "all"] - projected fields.
+   * @param {Array} [sortedData = "all"] - sorted field.
+   * @param {Array} [skipFields = "none"] - field not to include
+   * @returns {Array} List of survey submissions. 
+   */
+  
+  static surveySubmissionDocuments(
+    surveySubmissionFilter = "all", 
+    fieldsArray = "all",
+    sortedData = "all",
+    skipFields = "none"
+  ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+    
+            let queryObject = (surveySubmissionFilter != "all") ? surveySubmissionFilter : {};
+    
+            let projection = {}
+    
+            if (fieldsArray != "all") {
+                fieldsArray.forEach(field => {
+                    projection[field] = 1;
+                });
+            }
+
+            if( skipFields !== "none" ) {
+              skipFields.forEach(field=>{
+                projection[field] = 0;
+              })
+            }
+
+            let surveySubmissionDocuments;
+
+            if ( sortedData !== "all" ) {
+                
+                surveySubmissionDocuments = 
+                await database.models.surveySubmissions.find(
+                    queryObject, 
+                    projection
+                ).sort(sortedData).lean();
+
+            } else {
+                
+                surveySubmissionDocuments = 
+                await database.models.surveySubmissions.find(
+                    queryObject, 
+                    projection
+                ).lean();
+            }   
+
+            return resolve(surveySubmissionDocuments);
+            
+        } catch (error) {
+            return resolve({
+                success: false,
+                message: error.message,
+                data: false
+            })
+        }
+    });
+ }
+
+
+
+    /**
+    * Make survey submission.
+    * @method
+    * @name make
+    * @param {Object} req - request details 
+    * @returns {String} -  message.
+    */
+
+   static make( req = {} ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (Object.keys(req).length == 0) {
+                throw new Error(messageConstants.apiResponses.REQUEST_DETAILS_REQUIRED)
+            }
+
+            let surveySubmissionDocument = await this.surveySubmissionDocuments
+            (
+                { 
+                  _id : req.params._id,
+                  createdBy: req.userDetails.userId,
+                  status: messageConstants.common.SUBMISSION_STATUS_COMPLETED
+                }
+            )
+
+            if (surveySubmissionDocument.length > 0) {
+                throw new Error(messageConstants.apiResponses.MULTIPLE_SUBMISSIONS_NOT_ALLOWED)
+            }
+           
+            let response = await submissionsHelper.createEvidencesInSubmission
+            (  
+                req,
+                "surveySubmissions", 
```

Comment:
@deepakr123 - Create constants to be used across.

Reply:
resolved